src/bin/pg_autoctl/cli_do_show.c

```diff
@@ -0,0 +1,235 @@
+#include "cli_do_root.h"
+#include "commandline.h"
+#include "config.h"
+#include "defaults.h"
+#include "file_utils.h"
+#include "fsm.h"
+#include "ipaddr.h"
+#include "keeper_config.h"
+#include "keeper.h"
+#include "monitor.h"
+#include "monitor_config.h"
+#include "pgctl.h"
+#include "primary_standby.h"
+
+
+static void cli_show_ipaddr(int argc, char **argv);
+static void cli_show_cidr(int argc, char **argv);
+static void cli_show_lookup(int argc, char **argv);
+static void cli_show_nodename(int argc, char **argv);
+
+static CommandLine do_show_ipaddr_command =
+	make_command("ipaddr",
+				 "Print this node's IP address information", "", "",
+				 NULL, cli_show_ipaddr);
+
+static CommandLine do_show_cidr_command =
+	make_command("cidr",
+				 "Print this node's CIDR information", "", "",
+				 NULL, cli_show_cidr);
+
+static CommandLine do_show_lookup_command =
+	make_command("lookup",
+				 "Print this node's DNS lookup information",
+				 "<nodename>", "",
+				 NULL, cli_show_lookup);
+
+static CommandLine do_show_nodename_command =
+	make_command("nodename",
+				 "Print this node's default nodename", "", "",
+				 NULL, cli_show_nodename);
+
+CommandLine *do_show_subcommands[] = {
+ 	&do_show_ipaddr_command,
+	&do_show_cidr_command,
+	&do_show_lookup_command,
+	&do_show_nodename_command,
+	NULL
+};
+
+CommandLine do_show_commands =
+	make_command_set("show",
+					 "Show some debug level information", NULL, NULL,
+					 NULL, do_show_subcommands);
+
+
+
+/*
+ * cli_show_ipaddr displays the LAN IP address of the current node, as used
+ * when computing the CIDR address range to open in the HBA file.
+ */
+static void
+cli_show_ipaddr(int argc, char **argv)
+{
+	char ipAddr[BUFSIZE];
+
+	if (!fetchLocalIPAddress(ipAddr, BUFSIZE,
+							 PG_AUTOCTL_DEFAULT_SERVICE_NAME,
+							 PG_AUTOCTL_DEFAULT_SERVICE_PORT))
+	{
+		log_warn("Failed to determine network configuration.");
+		exit(EXIT_CODE_INTERNAL_ERROR);
+	}
+
+	fprintf(stdout, "%s\n", ipAddr);
+}
+
+
+/*
+ * cli_show_cidr displays the LAN CIDR that pg_autoctl grants connections to in
+ * the HBA file for setting up Postgres streaming replication and connections
+ * to the monitor.
+ */
+static void
+cli_show_cidr(int argc, char **argv)
+{
+	char ipAddr[BUFSIZE];
+	char cidr[BUFSIZE];
+
+	if (!fetchLocalIPAddress(ipAddr, BUFSIZE,
+							 PG_AUTOCTL_DEFAULT_SERVICE_NAME,
+							 PG_AUTOCTL_DEFAULT_SERVICE_PORT))
+	{
+		log_warn("Failed to determine network configuration.");
+		exit(EXIT_CODE_INTERNAL_ERROR);
+	}
+
+	if (!fetchLocalCIDR(ipAddr, cidr, BUFSIZE))
+	{
+		log_warn("Failed to determine network configuration.");
+		exit(EXIT_CODE_INTERNAL_ERROR);
+	}
+
+	fprintf(stdout, "%s\n", cidr);
+}
+
+
+/*
+ * cli_check_nodename checks that the --nodename argument is either an IP
+ * address that exists on the local list of interfaces, or a hostname that a
+ * DNS lookup solves to an IP address we have on the local machine.
+ *
+ * TODO: then do a reverse lookup of that IP address!
```

Comment:
is this still a todo? Seems like you already perform a lookup based on an ip address.

If it is still a TODO can we implement this already in the current PR instead of leaving the todo hanging here?

Reply:
I don't think there's a TODO left with the revision I ended up sending. Removed.